lib/validate.js

```diff
@@ -117,104 +117,121 @@ module.exports = {
-        const entry = path.relative('.', value);
-        const entryFile = _.replace(entry, new RegExp(`${path.extname(entry)}$`), '');
+        // Lookup associated Serverless functions
+        const allEntryFunctions = _.map(
+          this.serverless.service.getAllFunctions(),
+          funcName => {
+            const func = this.serverless.service.getFunction(funcName);
+            const handler = func.handler;
+            const handlerFile = path.relative('.', getHandlerFile(handler));
+            return {
+              handlerFile,
+              funcName,
+              func
+            };
+          }
+        );
+
+        this.entryFunctions = _.flatMap(entries, (value, key) => {
+          const entry = path.relative('.', value);
+          const entryFile = _.replace(entry, new RegExp(`${path.extname(entry)}$`), '');
+
+          const entryFuncs = _.filter(allEntryFunctions, [ 'handlerFile', entryFile ]);
+          if (_.isEmpty(entryFuncs)) {
+            // We have to make sure that for each entry there is an entry function item.
+            entryFuncs.push({});
+          }
+          _.forEach(entryFuncs, entryFunc => {
+            entryFunc.entry = {
+              key,
+              value
+            };
+          });
+          return entryFuncs;
+        });
 
-        const entryFuncs = _.filter(allEntryFunctions, [ 'handlerFile', entryFile ]);
-        if (_.isEmpty(entryFuncs)) {
-          // We have to make sure that for each entry there is an entry function item.
-          entryFuncs.push({});
-        }
-        _.forEach(entryFuncs, entryFunc => {
-          entryFunc.entry = {
-            key,
-            value
+        this.webpackConfig = _.map(this.entryFunctions, entryFunc => {
+          const config = _.cloneDeep(this.webpackConfig);
+          config.entry = {
+            [entryFunc.entry.key]: entryFunc.entry.value
           };
+          const compileName = entryFunc.funcName || _.camelCase(entryFunc.entry.key);
+          config.output.path = path.join(config.output.path, compileName);
+          return config;
         });
-        return entryFuncs;
-      });
+      } else {
+        this.webpackConfig.output.path = path.join(this.webpackConfig.output.path, 'service');
+      }
 
-      this.webpackConfig = _.map(this.entryFunctions, entryFunc => {
-        const config = _.cloneDeep(this.webpackConfig);
-        config.entry = {
-          [entryFunc.entry.key]: entryFunc.entry.value
-        };
-        const compileName = entryFunc.funcName || _.camelCase(entryFunc.entry.key);
-        config.output.path = path.join(config.output.path, compileName);
-        return config;
+      return BbPromise.resolve();
+    };
+    
+    // Webpack config can be a Promise, If it's a Promise wait for resolved config object.
+    if (this.webpackConfig && this.webpackConfig instanceof Promise) {
+      return new BbPromise((resolve, reject) => {
```

Comment:
We should remove the `new Promise` anti-pattern here.
```
return this.webpackConfig.then(processConfig);
```

Reply:
Copied.